packages/flutter/test/material/elevated_button_test.dart

```diff
@@ -1914,6 +1914,52 @@ void main() {
     expect(controller.value, <MaterialState>{MaterialState.disabled});
     expect(count, 1);
   });
+
+  testWidgetsWithLeakTracking('ElevatedButton.icon honors iconAlignment', (WidgetTester tester) async {
```

Comment:
```suggestion
  testWidgetsWithLeakTracking('ElevatedButton.icon honors iconAlignment', (WidgetTester tester) async {
```
Split this test into two,  "Default iconAlignment" and "iconAlignment can be customized" (it should  test both start and end position, in both text directions). 

Same for other test classes

Reply:
I'm confused about the textDirection still.

Reply:
Implemented

Reply:
Reimplemented